ibm/service/codeengine/resource_ibm_code_engine_function_test.go

```diff
@@ -0,0 +1,251 @@
+					resource.TestCheckResourceAttr("ibm_code_engine_function.code_engine_function_instance", "name", functionName),
+					resource.TestCheckResourceAttr("ibm_code_engine_function.code_engine_function_instance", "runtime", functionRuntime),
+					resource.TestCheckResourceAttr("ibm_code_engine_function.code_engine_function_instance", "code_binary", "false"),
+					resource.TestCheckResourceAttr("ibm_code_engine_function.code_engine_function_instance", "code_reference", functionCodeReferenceUpdate),
+					resource.TestCheckResourceAttr("ibm_code_engine_function.code_engine_function_instance", "managed_domain_mappings", functionManagedDomainMappingsUpdate),
+					resource.TestCheckResourceAttr("ibm_code_engine_function.code_engine_function_instance", "scale_concurrency", "1"),
+					resource.TestCheckResourceAttr("ibm_code_engine_function.code_engine_function_instance", "scale_cpu_limit", functionScaleCpuLimitUpdate),
+					resource.TestCheckResourceAttr("ibm_code_engine_function.code_engine_function_instance", "scale_down_delay", functionScaleDownDelayUpdate),
+					resource.TestCheckResourceAttr("ibm_code_engine_function.code_engine_function_instance", "scale_max_execution_time", functionScaleMaxExecutionTimeUpdate),
+					resource.TestCheckResourceAttr("ibm_code_engine_function.code_engine_function_instance", "scale_memory_limit", functionScaleMemoryLimitUpdate),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIbmCodeEngineFunctionConfigBasic(projectID string, codeReference string, name string, runtime string) string {
+	return fmt.Sprintf(`
+		data "ibm_code_engine_project" "code_engine_project_instance" {
+			project_id = "%s"
+		}
+
+		resource "ibm_code_engine_function" "code_engine_function_instance" {
+			project_id = data.ibm_code_engine_project.code_engine_project_instance.project_id
+			code_reference = "%s"
+			name = "%s"
+			runtime = "%s"
+
+			lifecycle {
+				ignore_changes = [
+					run_env_variables
+				]
+			}
+		}
+	`, projectID, codeReference, name, runtime)
+}
+
+func testAccCheckIbmCodeEngineFunctionConfig(projectID string, codeReference string, managedDomainMappings string, name string, runtime string, scaleCpuLimit string, scaleDownDelay string, scaleMaxExecutionTime string, scaleMemoryLimit string) string {
+	return fmt.Sprintf(`
+		data "ibm_code_engine_project" "code_engine_project_instance" {
+			project_id = "%s"
+		}
+
+		resource "ibm_code_engine_function" "code_engine_function_instance" {
+			project_id = data.ibm_code_engine_project.code_engine_project_instance.project_id
+			code_reference = "%s"
+			managed_domain_mappings = "%s"
+			name = "%s"
+			runtime = "%s"
+			scale_cpu_limit = "%s"
+			scale_down_delay = %s
+			scale_max_execution_time = %s
+			scale_memory_limit = "%s"
+            run_env_variables {
+				type  = "literal"
+				name  = "name"
+				value = "value"
+			}
+
+			lifecycle {
```

Comment:
This block is n't required when we use TypeSet. 
Adding this block restricts user's to do update of `run_env_variables`

Reply:
Thanks you very much for reviewing our PR. I have a question from our side though. We ignore changes here, because our system injects some `run_env_variables` on the server-side, resulting in diffs during testing and obviously from the user's perspective. Users can react by adding those system-generated variables to the definition alongside the user-defined ones.
Are you saying that if we changed it to `TypeSet`, Terraform would be able to identify changes in the user-set variables only and ignore outside changes?

Reply:
No TypeSet doesn't identify the changes . Can we loop , identify system-generated variables before setting back in read method?
Typeset is unordered list so it will help in controlling diff on the order of items in the list